### PR TITLE
wip: add archlinux test-scenario as platform

### PR DESCRIPTION
**Pull request by stefangweichinger**
new draft for getting the testing of arch linux into the default scenario

Tests OK locally here, I assume gh-actions to fail at first (workflow might need adjustments)

### CODE REVIEW

**Comment by stefangweichinger** [comment]
I don't know yet how to target that platform in the tests now ...

Locally I can do a `molecule test --platform-name rclone-archlinux` now successfully (which is nice IMO)

**Comment by gamezftw** [comment]
Take a look at this [link](https://ansible.readthedocs.io/projects/molecule/getting-started/?h=platform#run-a-full-test-sequence)
We can use `MOLECULE_PLATFORM_NAME` env variable to pass name of the platform.

Or [here](https://ansible.readthedocs.io/projects/molecule/getting-started/?h=platform#run-a-full-test-sequence), we can override `command` with `command: "test --platform rclone-arch"`.

I think env variable one would be nicer, what do you think?

**Comment by gamezftw** [comment]
I think [this comment](https://github.com/stefangweichinger/ansible-rclone/pull/162#issuecomment-2266565019) can be resolved here.

**Comment by stefangweichinger** [comment]
> Take a look at this [link](https://ansible.readthedocs.io/projects/molecule/getting-started/?h=platform#run-a-full-test-sequence) We can use `MOLECULE_PLATFORM_NAME` env variable to pass name of the platform.
> 
> Or [here](https://github.com/robertdebock/molecule-action?tab=readme-ov-file#command) we can override `command` with `command: "test --platform rclone-archlinux"`.
> 
> I think env variable one would be nicer, what do you think?

I agree .. might try that before leaving house for the weekend ... thanks for the pointer

**Comment by gamezftw** [comment]
I don't understand why [this step](https://github.com/stefangweichinger/ansible-rclone/actions/runs/10226387223/job/28296571024?pr=164#step:5:163) is passing without updating cache here and failing for me locally.... Do you think we should still add it?

**Comment by stefangweichinger** [comment]
looks good! @hkcomori pls check if that archlinux testing is ok in the gh actions now.

File: .github/workflows/molecule.yml [hunk 1]
```diff
@@ -90,10 +88,11 @@ jobs:
       - name: molecule
         uses: robertdebock/molecule-action@6.0.1
         env:
-          MOLECULE_DISTRO: "${{ matrix.distros }}"
           TEST_RELEASE: "${{ matrix.releases }}"
           TEST_VERSION: "${{ matrix.versions }}"
+          MOLECULE_PLATFORM_NAME: rclone-archlinux
         with:
+          image: carlodepieri/docker-archlinux-ansible
```
Comment (gamezftw) [inline .github/workflows/molecule.yml:95]: We should be able to remove this since it is configured in platform.

Comment (stefangweichinger) [reply .github/workflows/molecule.yml:95]: correct, that was just some random try yesterday

Comment (stefangweichinger) [reply .github/workflows/molecule.yml]: while we're at it: I think `galaxy_release.yml` in the workflows could be removed also. Was a try back then, the task is done within `molecule.yml`. Agree?

**Comment by hkcomori** [comment]
I rebase #162 to #164, the gh action is successful in #162.

**Comment by stefangweichinger** [comment]
looks good to me. pls let's wait a bit before merging, I 'd like to review things once again on monday or so. ok?
#164 and the changes to one scenario only, with 2 platforms, looks like a good solution to me.
Now we have to only maintain one molecule-scenario. Agree?

**Comment by hkcomori** [comment]
Yes, I agree.
In my opinion, one molecule-scenario is best unless there is a very special reason.

**Comment by stefangweichinger** [comment]
> Yes, I agree. In my opinion, one molecule-scenario is best unless there is a very special reason.

Yes. I noticed that the docker image the archlinux tests use is a bit old, is that true? Does it matter?

I think I might merge #164 right now before leaving for today. #162 maybe tomorrow. Just being cautious.

**Pull request merged**